target/linux/ramips/image/mt7621.mk

```diff
@@ -262,6 +262,15 @@ define Device/totolink_a7000r
 endef
 TARGET_DEVICES += totolink_a7000r
 
+define Device/adslr_g7
+  DTS := ADSLR-G7
+  IMAGE_SIZE := $(ralink_default_fw_size_16M)
```

Comment:
The size of your firmware partition is 0xf20000 = 15488k, not ralink_default_fw_size_16M (which is 15744k).
Change this line to:
```make
  IMAGE_SIZE := 15488k
```
instead.

Reply:
@981213 Concerning IMAGE_SIZE, just FYI: I've recently found that this is also wrong for many devices in ramips that just default (intentionally or not) to ralink_default_fw_size_8M . I will send a PR for that at some point, but still hope that someone gets interested in my ramips rename patch beforehand. I also think about dropping the default 8M (and moving it to devices instead), as it is arbitrary and actually invites you to overlook caring about the real value.